tap_zendesk/client.py

```diff
@@ -129,8 +121,135 @@ def parse_response(self, response: requests.Response) -> Iterable[dict]:
+
+    def _request(
+        self,
+        prepared_request: PreparedRequest,
+        context: Optional[Dict[str, Any]] = None,
+    ) -> Response:
+        """Send a HTTP request with rate limiting.
+
+        Args:
+            prepared_request: The prepared HTTP request.
+            context: Stream partition or context dictionary.
+
+        Returns:
+            The HTTP response object.
+        """
+        # Calculate the time difference between the current time and the last request time
+        time_since_last_request = time.time() - self.last_request_time
+
+        # Ensure we don't exceed 250 requests per minute (1 request per 0.24 seconds)
+        if time_since_last_request < 0.24:
+            time.sleep(0.24 - time_since_last_request)
+
+        response = self.requests_session.send(
+            prepared_request,
+            timeout=self.timeout,
+            allow_redirects=self.allow_redirects,
+        )
+
+        self.last_request_time = time.time()
+
+        self._write_request_duration_log(
+            endpoint=self.path,
+            response=response,
+            context=context,
+            extra_tags={"url": prepared_request.path_url}
+            if self._LOG_REQUEST_METRIC_URLS
+            else None,
+        )
+
+        # Check for 404 Not Found and log a warning
+        if response.status_code == 404:
+            self.logger.warning(f"Received 404 for URL: {response.url}")
+            return None
+
+        self.validate_response(response)
+        self.logger.debug("Response received successfully.")
+        return response
+
+    def get_records(self, context):
+        end_date_str = self.config.get("end_date")
+        end_date = datetime.fromisoformat(end_date_str) if end_date_str else None
+
+        if end_date and end_date.tzinfo is None:
+            end_date = end_date.replace(tzinfo=timezone.utc)
+
+        replication_key = self.replication_key
+
+        paginator = self.get_new_paginator()
+        next_page_token = None
+
+        while True:
+            prepared_request = self.prepare_request(context, next_page_token)
+            response = self._request(prepared_request, context)
+            if not response:
+                self.logger.error("Received empty response")
+                break
+
+            records = self.parse_response(response)
+
+            for record in records:
+                if not record:
+                    self.logger.error("Received empty record")
+                    continue
+
+                try:
+                    record_json = json.loads(json.dumps(record))
+                except json.JSONDecodeError as e:
+                    self.logger.error(f"JSONDecodeError encountered while parsing record: {e}")
+                    continue
+
+                record_date_str = record.get(replication_key)
+                if record_date_str:
+                    self.logger.debug(record_date_str)
```

Comment:
if you run any tap in debug loglevel you will see the record is already printed, so no need to log again.